components/ipfs/ipfs_utils.cc

```diff
@@ -460,47 +464,110 @@ std::string GetRegistryDomainFromIPNS(const GURL& url) {
       cid, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
 }
 
-// gateway.io/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy ->
-// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
-// bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy.ipfs.gateway.io ->
-// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
-absl::optional<GURL> TranslateToCurrentGatewayUrl(const GURL& url) {
-  if (!url.is_valid() || !url.SchemeIsHTTPOrHTTPS()) {
-    return absl::nullopt;
+std::string DecodeSingleLabelForm(const std::string& input) {
+  if (input.find('.') != std::string::npos) {
+    return input;
+  }
```

Comment:
why this check?
we already supposed to have `input` as part of string split by dot

Reply:
In case when IPNS url is a part of the URL's path it can be in both single-label and normal form.

Reply:
Add a comment here to say:
`Normal form URLs such as en.wikipedia-on-ipfs.org should stay as is.`